src/config.rs

```diff
@@ -0,0 +1,219 @@
+use std::{
+    fs::{self, File},
+    env,
+    path::Path,
+    io::Write,
+    process::exit,
+    sync::{Arc, Mutex},
+};
+use regex::Regex;
+use serde::Deserialize;
+use toml::value;
+
+/// The response that is injected into a panic, if the config file was configured falsely
+const PANIC_RESPONSE: &str = "Please create a config file yourself or try setting the environment CONFIG_FILE to valid file location!";
+
+/// The versions of the config this program is compatible with
+const COMPATIBLE_VERSIONS: [(u32, u32); 1] = [
+    (0, 2),
+];
+
+lazy_static! {
+    ///
+    /// The global, thread safe configuration of all of this bot.
+    ///
+    /// Examples:
+    /// ```
+    /// let config_arc = Arc::clone(&CONFIG);
+    /// let mut config_lock = config_arc.lock();
+    /// let value = match config_lock {
+    ///     Ok(config) => {
+    ///         // ACCESS CONFIG FIELDS HERE AND COPY THEM INTO VALUE
+    ///         // Example: String::from(&config.autokommentator.token)
+    ///     },
+    ///     Err(why) => {
+    ///         panic!("Something went wrong internally: {:?}\nMutex is poisoned: {}", why, why);
+    ///     }
+    /// };
+    /// ```
+    ///
+    pub static ref CONFIG: Arc<Mutex<Config>> = Arc::new(Mutex::new(Config::new()));
```

Comment:
Put the regex always in a lazy_static
```suggestion
    pub static ref CONFIG: Arc<Mutex<Config>> = Arc::new(Mutex::new(Config::new()));

    static ref VERSION_REGEX: Regex = Regex::new("(?m)^version *= *\"(?P<version>\\d*\\.\\d*)\" *(|#.*)$").unwrap();
```